src/engine/enginebuffer.cpp

```diff
@@ -929,18 +930,13 @@ void EngineBuffer::process(CSAMPLE* pOutput, const int iBufferSize) {
         if (!bCurBufferPaused) {
             // The fileposition should be: (why is this thing a double anyway!?
             // Integer valued.
-            double filepos_play_rounded = round(m_filepos_play);
-            if (filepos_play_rounded != m_filepos_play) {
-                qWarning() << __FILE__ << __LINE__ << "ERROR: filepos_play is not round:" << m_filepos_play;
+            double playFrame = m_filepos_play / kSamplesPerFrame;
+            double filepos_play_rounded = round(playFrame) * kSamplesPerFrame;
```

Comment:
why divide and multiply when you can just do the same mod and add we've been using?  Also this idiom is repeated a few times and probably exists elsewhere in the code.  Factor it out into an inlined function in sample_util called roundToFrame(position);